monai/transforms/post/array.py

```diff
@@ -679,17 +679,21 @@ def __init__(self, weights: Sequence[float] | NdarrayOrTensor | None = None) ->
         self.weights = torch.as_tensor(weights, dtype=torch.float) if weights is not None else None
 
     def __call__(self, img: Sequence[NdarrayOrTensor] | NdarrayOrTensor) -> NdarrayOrTensor:
-        img_ = self.get_stacked_torch(img)
-        if self.weights is not None:
-            self.weights = self.weights.to(img_.device)
-            shape = tuple(self.weights.shape)
-            for _ in range(img_.ndimension() - self.weights.ndimension()):
-                shape += (1,)
-            weights = self.weights.reshape(*shape)
+        out_pt = None
+        total_weight = 0.0
+
+        for i, pred in enumerate(img):
+            pred = torch.as_tensor(pred)
+            if out_pt is None:
+                out_pt = torch.zeros_like(pred)
+
+            if self.weights is not None:
+                weight = self.weights[i].to(pred.device)
 
-            img_ = img_ * weights / weights.mean(dim=0, keepdim=True)
+            out_pt += pred * weight
```

Comment:
If `self.weights` is None, this line will show error. You may need set a default value or move the computation in the for-loop.